### PR TITLE
test build - 150 browserstack

### LINKED ISSUE #150: Setup BrowserStack, Acceptance Testing
- [x] configure karma
- [x] run karma tests across browsers in BrowserStack
- [x] setup a "kitchen sink" docs page for screenshot testing
- [ ] run screenshots in BrowserStack

### PR DESCRIPTION
**Pull request by sodiumjoe**
Fixes #150

## PR Checklist

- ~~Manually tested across supported browsers~~
- [x] ~~Unit tests written (`common` at minimum)~~
- [x] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [x] ~~One core team UX approval~~

### CODE REVIEW

File: browserstack.conf.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module.exports = {
+	bs_chrome: {
```
Comment (jondlm) [inline browserstack.conf.js]: Looks like these are repeated in 3 places. Might it make sense to pull these out into their own file? I'm fine either way honestly.

File: package.json [hunk 2]
```diff
@@ -11,10 +11,15 @@
     "dev": "gulp docs-generate && webpack-dev-server --inline --content-base src/docs/",
     "docs-upload-gh": "./scripts/docs-upload-gh.sh",
     "docs-upload": "gulp docs-upload",
-    "preversion": "rm -rf dist && TZ=UTC gulp preversion",
+    "preversion": "rm -rf dist && npm run test && gulp preversion",
     "postversion": "gulp docs-upload",
-    "test": "rm -rf dist && TZ=UTC gulp lint test",
-    "test-tdd": "rm -rf dist && TZ=UTC gulp test-tdd"
+    "start": "npm run dev",
```
Comment (jondlm) [inline package.json]: Haha, thank you 😄

File: package.json [hunk 3]
```diff
@@ -42,9 +47,11 @@
     "babel-core": "^6.4.5",
     "babel-loader": "^6.2.1",
     "babel-plugin-rewire": "^1.0.0-rc-4",
+    "babel-preset-airbnb": "^2.0.0",
```
Comment (jondlm) [inline package.json]: We going with airbnb stuff now?

File: scripts/env-check.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+if [ -z $BROWSERSTACK_USERNAME ]; then
```
Comment (jondlm) [inline scripts/env-check.sh]: Noice

File: scripts/local.runner.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+});
+const local = new browserstack.Local();
+Nightwatch.bs_local = local;
+const browserStackConfig = {
+	key: process.env.BROWSERSTACK_ACCESS_KEY,
+};
+
+try {
+
+	process.mainModule.filename = './node_modules/.bin/nightwatch'
```
Comment (jondlm) [inline scripts/local.runner.js]: _shiver_

File: browserstack.conf.js (hunk 1 above)
Comment (sodiumjoe) [inline browserstack.conf.js]: i actually did initially, but there's slight differences like "ie" vs "internet_explorer" between configs for selenium vs karma. 😢

File: src/components/Accordion/Accordion.spec.jsx [hunk 6]
```diff
@@ -102,19 +101,29 @@ describe('Accordion', () => {
 	});
 });
 
-describeWithDOM('Accordion', () => {
```
Comment (jondlm) [inline src/components/Accordion/Accordion.spec.jsx]: Are we doing away with describewithdom altogether?

**Comment by jondlm** [comment]
Just did a once over, LGTM! Nice work so far on this.

File: browserstack.conf.js (hunk 1 above)
Comment (jondlm) [inline browserstack.conf.js]: Ahh okay, makes sense. I figured it was something like that.

File: package.json (hunk 3 above)
Comment (sodiumjoe) [inline package.json]: not used by us directly, it's used in the webpack config to bundle enzyme

File: src/components/Accordion/Accordion.spec.jsx (hunk 6 above)
Comment (sodiumjoe) [inline src/components/Accordion/Accordion.spec.jsx]: don't need it since we have a real dom

**Comment by ogupte** [comment]
i'm not a fan of committing all the screenshots for the threshold comparison. the size of the repo has significantly ballooned, and i'm not sure if it's worth it.

**Comment by ogupte** [comment]
this should be removed:
https://github.com/appnexus/lucid/blob/150-browserstack/src/components/DropMenu/DropMenu.spec.jsx#L636

**Comment by jondlm** [comment]
That's a good point @ogupte -- I don't like having binary files in our repo either :/ Doesn't github have some files storage mechanism for these kinds of things? I've never used it but I remember reading about it a while back.

https://github.com/github/git-lfs

**Comment by ogupte** [comment]
this should be using `omitProps`:
https://github.com/appnexus/lucid/blob/150-browserstack/src/components/Icon/DataViewIcon/DataViewIcon.jsx#L30

**Comment by ogupte** [comment]
If Icon now has `isClickable` doesn't it follow that it should also have `onClick`?
https://github.com/appnexus/lucid/blob/150-browserstack/src/components/Icon/Icon.jsx#L57

**Comment by ogupte** [comment]
sry i was doing `git diff HEAD~35` which included all the icons stuff...ignore those comments

**Comment by ogupte** [comment]
in the overlay test (https://github.com/appnexus/lucid/blob/150-browserstack/src/components/Overlay/Overlay.spec.jsx#L132) you should be using the constant defined at https://github.com/appnexus/lucid/blob/150-browserstack/src/constants/key-code.js#L8 for the keycode

**Comment by ogupte** [comment]
`{dX: dX}` can be `{dX}`:
https://github.com/appnexus/lucid/blob/150-browserstack/src/components/SplitVertical/SplitVertical.spec.jsx#L205

**Comment by sodiumjoe** [comment]
Yeah, i thought about looking into git-annex or something, but i didn't want to spend anymore time on this. Might be wise to punt on it for the moment, we can leave all the infrastructure and just remove the screenshot tests from `npm test`, and delete the screenshot binaries for now. In the future it might be better to go through and actually enumerate the examples that we want to screenshot instead of doing all of them and excluding a few, that was just the fastest thing for me to do.

**Pull request merged**